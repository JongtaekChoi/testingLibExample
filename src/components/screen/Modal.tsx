import React, { ReactElement, useRef } from 'react';

import { DefaultNavigationProps } from '../../types';
import styled from 'styled-components/native';
import Modal from 'react-native-modalbox';
import Button from '../shared/Button';

const Container = styled.View`
  flex: 1;
  background-color: transparent;
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const StyledText = styled.Text`
  font-size: 16;
  color: blue;
`;

interface Props {
  navigation: DefaultNavigationProps<'default'>;
}

function Page(props: Props): ReactElement {
  const modal: React.MutableRefObject<Modal | null>  = useRef({})
  const open = (): void => {
    if(modal.current) {
      modal.current.open();
    }
  };
  const close = (): void => {
    if(modal.current) {
      modal.current.close();
    }
  };
  return (
    <Container>
      <Button
        testID="openBtn"
        onClick={open}
        text='열기'/> 
      <Modal ref={modal}>
          <Button
            testID="closeBtn"
            onClick={close}
            text='닫기'
          />
      </Modal>
    </Container>
  );
}

export default Page;
