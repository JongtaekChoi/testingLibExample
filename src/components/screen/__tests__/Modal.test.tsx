import 'react-native';

import React, { ReactElement } from 'react';
import { RenderResult, render } from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Modal from '../Modal';
import renderer from 'react-test-renderer';

let props: any;
let component: ReactElement;
let testingLib: RenderResult;

describe('[Modal] screen', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Modal {...props} />);
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onClick', () => {
      
    });
  });
});
