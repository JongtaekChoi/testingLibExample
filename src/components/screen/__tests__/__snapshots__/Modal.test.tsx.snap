// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Modal] screen renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="openBtn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#069ccd",
            "borderColor": "#069ccd",
            "borderRadius": 4,
            "borderWidth": 2,
            "height": 52,
            "justifyContent": "center",
            "width": 320,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 14,
            },
          ]
        }
      >
        열기
      </Text>
    </View>
  </View>
  <View />
</View>
`;
